src/Decode.fs

```diff
@@ -510,65 +496,70 @@ module Decode =
         fun path value ->
             if Helpers.isArray value then
                 let mutable i = -1
-                let tokens = Helpers.asArray value
-                (Ok [], tokens) ||> Array.fold (fun acc value ->
+                let mutable acc = Ok []
+
+                for token in value.EnumerateArray() do
```

Comment:
the MSDN docs suggest you avoid wrapping these enumerators in collections where possible, so I opted to use mutation and direct enumeration here